frame/support/procedural/src/construct_runtime/expand/call.rs

```diff
@@ -0,0 +1,145 @@
+		let index = pallet_declaration.index;
+
+		variant_defs.extend(quote!(#[codec(index = #index)] #name( #scrate::dispatch::CallableCallFor<#name, #runtime> ),));
+		variant_patterns.push(quote!(Call::#name(call)));
+		pallet_names.push(name);
+		query_call_part_macros.push(quote! {
+			#path::__substrate_call_check::is_call_part_defined!(#name);
+		});
+	}
+
+	quote! {
+		#( #query_call_part_macros )*
+
+		#[derive(
+			Clone, PartialEq, Eq,
+			#scrate::codec::Encode,
+			#scrate::codec::Decode,
+			#scrate::RuntimeDebug,
+		)]
+		pub enum Call {
+			#variant_defs
+		}
+		impl #scrate::dispatch::GetDispatchInfo for Call {
+			fn get_dispatch_info(&self) -> #scrate::dispatch::DispatchInfo {
+				match self {
+					#( #variant_patterns => call.get_dispatch_info(), )*
+				}
+			}
+		}
+		impl #scrate::dispatch::GetCallMetadata for Call {
+			fn get_call_metadata(&self) -> #scrate::dispatch::CallMetadata {
+				use #scrate::dispatch::GetCallName;
+				match self {
+					#(
+						#variant_patterns => {
+							let function_name = call.get_call_name();
+							let pallet_name = stringify!(#pallet_names);
+							#scrate::dispatch::CallMetadata { function_name, pallet_name }
+						}
+					)*
+				}
+			}
+
+			fn get_module_names() -> &'static [&'static str] {
+				&[#(
+					stringify!(#pallet_names),
+				)*]
+			}
+
+			fn get_call_names(module: &str) -> &'static [&'static str] {
+				use #scrate::dispatch::{Callable, GetCallName};
+				match module {
+					#(
+						stringify!(#pallet_names) =>
+							<<#pallet_names as Callable<#runtime>>::Call
+								as GetCallName>::get_call_names(),
+					)*
+					_ => unreachable!(),
+				}
+			}
+		}
+		impl #scrate::dispatch::Dispatchable for Call {
+			type Origin = Origin;
+			type Config = Call;
+			type Info = #scrate::weights::DispatchInfo;
+			type PostInfo = #scrate::weights::PostDispatchInfo;
+			fn dispatch(self, origin: Origin) -> #scrate::dispatch::DispatchResultWithPostInfo {
+				if !<Self::Origin as #scrate::traits::OriginTrait>::filter_call(&origin, &self) {
+					return #scrate::sp_std::result::Result::Err(#scrate::dispatch::DispatchError::BadOrigin.into());
+				}
+
+				#scrate::traits::UnfilteredDispatchable::dispatch_bypass_filter(self, origin)
+			}
+		}
+		impl #scrate::traits::UnfilteredDispatchable for Call {
+			type Origin = Origin;
+			fn dispatch_bypass_filter(self, origin: Origin) -> #scrate::dispatch::DispatchResultWithPostInfo {
+				match self {
+					#(
+						#variant_patterns =>
+							#scrate::traits::UnfilteredDispatchable::dispatch_bypass_filter(call, origin),
+					)*
+				}
+			}
+		}
+
+		#(
+			impl #scrate::traits::IsSubType<#scrate::dispatch::CallableCallFor<#pallet_names, #runtime>> for Call {
+				#[allow(unreachable_patterns)]
+				fn is_sub_type(&self) -> Option<&#scrate::dispatch::CallableCallFor<#pallet_names, #runtime>> {
+					match self {
+						#variant_patterns => Some(call),
+						// May be unreachable
+						_ => None,
```

Comment:
Why is it maybe?

Reply:
I simply copied and pasted code from `frame/support/src/call.rs`, but reading the code here, it may be the case where `variant_patterns` already contains an exhaustive list of the enum variants for `Call`, or it may not. I think this can happen in cases where a pallet has a defined `Call` part, but for whatever reason `construct_runtime` chose not to import said pallet's Call part, which would result in `variant_patterns` not being an exhaustive list.

Reply:
That makes sense